controller/platform_clients.go

```diff
@@ -0,0 +1,174 @@
+package controller
+
+import (
+	"context"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/kubernetes"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	errs "github.com/pkg/errors"
+	"net/url"
+	"os"
+)
+
+// ClientGetter creates an instances of clients used by this controller
+type ClientGetter interface {
```

Comment:
1. Could you use a more meaningful name please?
2. This is not something that should be part of the controller layer. This should be a part of the service layer. Could you maybe put in a different new/existing package?

Reply:
Yes, we tried to decouple it from the controller layer but stumbled into the extensive refactoring. Though it's better to do progressive refactoring.

Reply:
`GetAndCheckOSIOClient` is tightly coupled to the controller layer.